tests/operators/test_operators.py

```diff
@@ -128,80 +133,114 @@ def test_sum_operator_linearoperator():
     assert not isinstance(c, LinearOperator), 'Operator + LinearOperator should not be a LinearOperator'
 
 
-def test_elementwise_product_operator():
+@pytest.mark.parametrize('value', [2, 3j])
+def test_elementwise_product_operator(value):
     a = DummyOperator(torch.tensor(2.0))
-    b = torch.tensor(3.0)
+    b = torch.ones(10) * value
     c = a * b
-    x = torch.arange(10)
+    x = RandomGenerator(0).complex64_tensor(10)
     (y1,) = c(x)
-    y2 = a(x)[0] * b
+    (y2,) = a(b * x)
 
     torch.testing.assert_close(y1, y2)
     assert isinstance(c, Operator), 'Operator * scalar should be an Operator'
     assert not isinstance(c, LinearOperator), 'Operator * scalar should not be a LinearOperator'
 
 
-def test_elementwise_rproduct_operator():
+@pytest.mark.parametrize('value', [2, 3j])
+def test_elementwise_rproduct_operator(value):
     a = DummyOperator(torch.tensor(2.0))
-    b = torch.tensor(3.0)
+    b = torch.tensor(value)
     c = b * a
-    x = torch.arange(10)
+    x = RandomGenerator(0).complex64_tensor(10)
     (y1,) = c(x)
-    y2 = a(x)[0] * b
+    y2 = b * a(x)[0]
 
     torch.testing.assert_close(y1, y2)
     assert isinstance(c, Operator), 'Operator * scalar should be an Operator'
```

Comment:
To be super accurate this should be scaler * Operator and similarly for the other rproduct tests